src/main/java/fi/hsl/transitdata/vehicleposition/application/VehiclePositionHandler.java

```diff
@@ -99,21 +100,38 @@ private static String getUniqueVehicleId(int oper, int veh) {
     public void handleMessage(Message message) {
         try {
             if (TransitdataSchema.hasProtobufSchema(message, TransitdataProperties.ProtobufSchema.PassengerCount)) {
-                PassengerCount.Data data = PassengerCount.Data.parseFrom(message.getData());
-
+                log.info("Starting to handle passenger count data");
+                PassengerCount.Data data = null;
+                try {
+                    data = PassengerCount.Data.parseFrom(message.getData());
+                } catch (InvalidProtocolBufferException e) {
+                    log.error("Failed to parse passenger count data", e);
+                    throw new Exception(e);
+                }
+    
                 final String uniqueVehicleId = getUniqueVehicleId(data.getPayload().getOper(), data.getPayload().getVeh());
 
                 passengerCountCache.updatePassengerCount(uniqueVehicleId, data.getPayload().getRoute(), data.getPayload().getOday(), data.getPayload().getStart(), data.getPayload().getDir(), data.getPayload());
+                log.info("Done handling passenger count data");
             } else if (TransitdataSchema.hasProtobufSchema(message, TransitdataProperties.ProtobufSchema.HfpData)) {
-                Hfp.Data data = Hfp.Data.parseFrom(message.getData());
-
+                log.info("Starting to handle HfpData");
+                Hfp.Data data = null;
+                try {
+                    data = Hfp.Data.parseFrom(message.getData());
+                } catch (InvalidProtocolBufferException e) {
+                    log.error("Failed to parse HfpData", e);
+                    throw new Exception(e);
+                }
+    
                 //Ignore HFP messages that are not sent from vehicles on a journey
                 if (data.getTopic().getJourneyType() != Hfp.Topic.JourneyType.journey) {
+                    log.info("Ignore HFP messages that are not sent from vehicles on a journey");
                     return;
                 }
 
                 //Ignore HFP messages that are not sent from vehicles on an ongoing journey
                 if (data.getTopic().getTemporalType() != Hfp.Topic.TemporalType.ongoing) {
+                    log.info("Ignore HFP messages that are not sent from vehicles on an ongoing journey");
                     return;
                 }
```

Comment:
```suggestion
                if (data.getTopic().getJourneyType() != Hfp.Topic.JourneyType.journey || data.getTopic().getTemporalType() != Hfp.Topic.TemporalType.ongoing) {
                    log.info("Ignore HFP messages that are not sent from vehicles on a journey");
                    return;
                }
```

Nää on samat blokit, voisi ehkä yhdistää niin on vähän kauniimpaa ja lyhyempää koodia